### PR TITLE
Scala TFPark optimizer

### PR DESCRIPTION
**Pull request by jenniew**
Scala TFPark optimizer.
Simple example and Readme.

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/tfpark/Train.scala [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        BytesToGreyImg(28, 28) ->
+        GreyImgNormalizer(trainMean * 255, trainStd * 255) ->
+        GreyImgToSample() ->
+        new SampleToTFTrainSample() ->
+        SampleToMiniBatch(params.batchSize)
+
+      val optimizer = new TFOptimizer(params.modelPath,
+        new Adam[Float](1e-3),
+        trainDataSet)
+      optimizer.optimize(Trigger.maxEpoch(params.numEpoch))
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/tfpark/Train.scala]: What's the result of the training? Maybe we provide a method to export a checkpoint?
See [TFTrainingHelper.saveCheckpoint](https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/tfpark/TFTrainingHelper.scala#L152)

File: zoo/src/main/scala/com/intel/analytics/zoo/tfpark/TFOptimizer.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.zoo.feature.FeatureSet
+import com.intel.analytics.zoo.pipeline.estimator.Estimator
+
+
+class TFOptimizer(modelPath: String,
+                  optimMethod: OptimMethod[Float],
+                  trainData: FeatureSet[MiniBatch[Float]],
+                  validationData: FeatureSet[MiniBatch[Float]] = null,
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/tfpark/TFOptimizer.scala]: Support ValidationMethod?

Comment (jenniew) [reply zoo/src/main/scala/com/intel/analytics/zoo/tfpark/TFOptimizer.scala]: validation will be added later.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/tfpark/Train.scala (hunk 1 above)
Comment (jenniew) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/tfpark/Train.scala]: ok. will add checkpoint

**Pull request closed without merging**